### PR TITLE
feat: convert solutionForm to ts

### PR DESCRIPTION
**Pull request by Sembauke**
Checklist:

- [x] I have read [freeCodeCamp's contribution guidelines](https://contribute.freecodecamp.org).
- [x] My pull request has a descriptive title (not a vague title like `Update index.md`)
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes #XXXXX

### CODE REVIEW

File: client/src/templates/Challenges/projects/SolutionForm.tsx [hunk 1]
```diff
@@ -10,25 +10,35 @@ import {
...
-export class SolutionForm extends Component {
-  constructor(props) {
+interface SubmitProps {
+  isShouldCompletionModalOpen: boolean;
+}
+
+interface FormProps extends WithTranslation {
+  challengeType: number;
+  description?: string;
+  isSubmitting?: boolean;
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/projects/SolutionForm.tsx:20]:
```suggestion
```
We can drop this - it's never used.

File: client/src/templates/Challenges/projects/SolutionForm.tsx [hunk 2]
```diff
@@ -93,11 +101,14 @@ export class SolutionForm extends Component {
...
-            ? 'https://colab.research.google.com/drive/1i5EmInTWi1RFvFr2_aRXky96YxY6sbWy'
-            : 'https://replit.com/@camperbot/hello');
+        if (description) {
+          solutionLink =
+            solutionLink +
+            (description.includes('Colaboratory')
+              ? 'https://colab.research.google.com/drive/1i5EmInTWi1RFvFr2_aRXky96YxY6sbWy'
+              : 'https://replit.com/@camperbot/hello');
+          break;
+        }
```
Comment (ojeytonwilliams) [inline client/src/templates/Challenges/projects/SolutionForm.tsx:111]:
```suggestion
        solutionLink =
          solutionLink +
          (description?.includes('Colaboratory')
            ? 'https://colab.research.google.com/drive/1i5EmInTWi1RFvFr2_aRXky96YxY6sbWy'
            : 'https://replit.com/@camperbot/hello');
```
Nitpick.

**Review by ojeytonwilliams — CHANGES_REQUESTED** [review]
Thanks for persevering with this, Sem.  Just two nitpicks:

**Review by ojeytonwilliams — DISMISSED** [review]
LGTM :+1:

**Review by ShaunSHamilton — CHANGES_REQUESTED** [review]
Sorry, but i am going to nitpick, and ask that the file name be changed to kebab. Otherwise, this LGTM

**Review by ShaunSHamilton — APPROVED** [review]
This LGTM 👍 . Thank you, for that change

**Review by ojeytonwilliams — APPROVED** [review]
LGTM 👍

**Pull request merged**